files/chef-marketplace-cookbooks/chef-marketplace/recipes/_upgrade_analytics.rb

```diff
@@ -1,9 +1,31 @@
+bash "opscode-analytics-ctl reconfigure" do
+  code "opscode-analytics-ctl reconfigure"
+  only_if { analytics_configured? }
+  action :nothing
+end
+
+ruby_block "remove-analytics-from-cache" do
+  block do
+    package = Dir["#{Chef::Config[:file_cache_path]}/*"].find { |f| f =~ /opscode-analytics/ }
+    FileUtils.rm(package) if package && File.exist?(package)
+  end
+  action :nothing
+end
+
+bash "opscode-analytics-ctl stop" do
+  code "opscode-analytics-ctl stop"
+  only_if { analytics_configured? }
+end
+
 chef_ingredient "analytics" do
   action :upgrade
-  notifies :run, "bash[opscode-analytics-ctl reconfigure]", :immediately if analytics_configured?
+  notifies :run, "bash[opscode-analytics-ctl reconfigure]", :immediately
+  notifies :run, "ruby_block[remove-analytics-from-cache]", :immediately
```

Comment:
Is this a technical necessity, or just a "clean-up-so-our-image-is-as-small-as-possible" step?